optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,456 @@
+        elif crossover_name == "undx":
+            sigma_xi = 0.5
+            sigma_eta = 0.35 / np.sqrt(len(xs[0]))
+            params_array = _undx(xs[0], xs[1], xs[2], rng, sigma_xi, sigma_eta)
+        elif crossover_name == "undxm":
+            _m = len(xs) - 2
+            _n = len(xs[0])
+            assert _n - _m > 0
+
+            sigma_xi = 1 / np.sqrt(_m)
+            sigma_eta = (
+                0.35
+                * np.sqrt(_m + 1)
+                * np.sqrt(3)
+                / np.sqrt(_n - _m)
+                / np.sqrt(_m + 2)
+                / np.sqrt(2)
+            )
+            params_array = _undxm(xs, rng, sigma_xi, sigma_eta)
+        elif crossover_name == "spx":
+            epsilon = np.sqrt(len(xs[0]) + 2)
+            params_array = _spx(xs, rng, epsilon)
+        else:
+            assert False
+
+        _params = [
+            trans.untransform(np.array([param])) for trans, param in zip(transes, params_array)
+        ]
+        for param in _params:
+            for param_name in param.keys():
+                child[param_name] = param[param_name]
+
+        if _is_contained(child, search_space):
+            break
+
+    return child
+
+
+def _selection(
+    crossover_name: str,
+    study: Study,
+    parent_population: Sequence[FrozenTrial],
+    rng: np.random.RandomState,
+    dominates: Callable[[FrozenTrial, FrozenTrial, Sequence[StudyDirection]], bool],
+) -> List[FrozenTrial]:
+    if crossover_name in ["uniform", "blxalpha", "sbx", "vsbx"]:
+        n_select = 2
+    elif crossover_name in ["undx", "spx"]:
+        n_select = 3
+    elif crossover_name in ["undxm"]:
+        n_select = 4
+    else:
+        assert False
+    if len(parent_population) < n_select:
+        raise ValueError(
+            f"Using {crossover_name},"
+            f"the population size should be greater than or equal to {n_select}."
+        )
+
+    parents = []
+    for _ in range(n_select):
+        parent = _select_parent(
+            study, [t for t in parent_population if t not in parents], rng, dominates
+        )
+        parents.append(parent)
+    return parents
+
+
+def _select_parent(
+    study: Study,
+    population: Sequence[FrozenTrial],
+    rng: np.random.RandomState,
+    dominates: Callable[[FrozenTrial, FrozenTrial, Sequence[StudyDirection]], bool],
+) -> FrozenTrial:
+    # TODO(ohta): Consider to allow users to specify the number of parent candidates.
+    population_size = len(population)
+    candidate0 = population[rng.choice(population_size)]
+    candidate1 = population[rng.choice(population_size)]
+
+    # TODO(ohta): Consider crowding distance.
+    if dominates(candidate0, candidate1, study.directions):
+        return candidate0
+    else:
+        return candidate1
+
+
+def _swap(p0_i: Any, p1_i: Any, rand: float, swapping_prob: float) -> Any:
+    if rand < swapping_prob:
+        return p1_i
+    else:
+        return p0_i
```

Comment:
I think we don't have to define this utility function, since we can write this in one line like `p1_i if rand < swapping_prob else p0_i`.